qiskit_machine_learning/circuit/library/qnn_circuit.py

```diff
@@ -0,0 +1,243 @@
+        #      │  ZZFeatureMap(x[0],x[1]) │»
+        # q_1: ┤1                         ├»
+        #      └──────────────────────────┘»
+        # «     ┌──────────────────────────────────────────────────────────┐
+        # «q_0: ┤0                                                         ├
+        # «     │  RealAmplitudes(θ[0],θ[1],θ[2],θ[3],θ[4],θ[5],θ[6],θ[7]) │
+        # «q_1: ┤1                                                         ├
+        # «     └──────────────────────────────────────────────────────────┘
+
+        print(qnn_qc.num_qubits)
+        # prints: 2
+
+        print(qnn_qc.input_parameters)
+        # prints: ParameterView([ParameterVectorElement(x[0]), ParameterVectorElement(x[1])])
+
+        print(qnn_qc.weight_parameters)
+        # prints: ParameterView([ParameterVectorElement(θ[0]), ParameterVectorElement(θ[1]),
+        #         ParameterVectorElement(θ[2]), ParameterVectorElement(θ[3]),
+        #         ParameterVectorElement(θ[4]), ParameterVectorElement(θ[5]),
+        #         ParameterVectorElement(θ[6]), ParameterVectorElement(θ[7])])
+    """
+
+    def __init__(
+        self,
+        num_qubits: int | None = None,
+        feature_map: QuantumCircuit | None = None,
+        ansatz: QuantumCircuit | None = None,
+    ) -> None:
+        """
+        Although all parameters default to None at least one parameter must be provided, to determine
+        the number of qubits from it, when the instance is created.
+
+        If more than one parameter is passed:
+        1) If num_qubits is provided the feature map and/or ansatz supplied will be overridden to
+        circuits with num_qubits, as long as the respective circuit supports updating its number of
+        qubits.
+        2) If num_qubits is not provided the feature_map and ansatz must be set to the same number
+        of qubits.
+
+        Args:
+            num_qubits:  Number of qubits, a positive integer. Optional if feature_map or ansatz is
+                         provided, otherwise required. If not provided num_qubits defaults from the
+                         sizes of feature_map and ansatz.
+            feature_map: A feature map. Optional if num_qubits or ansatz is provided, otherwise
+                         required. If not provided defaults to ``ZZFeatureMap`` or ``ZFeatureMap``
+                         if num_qubits is determined to be 1.
+            ansatz:      An ansatz. Optional if num_qubits or feature_map is provided, otherwise
+                         required. If not provided defaults to ``RealAmplitudes``.
+
+        Returns:
+            The composed feature map and the ansatz circuit.
+
+        Raises:
+            QiskitMachineLearningError: If the correct number of qubits cannot be derived form
+            the provided input arguments.
+            ValueError: If the number of qubits is invalid.
+            AttributeError: If the feature_map or ansatz is invalid.
```

Comment:
I don't know about listing AttributeError here. Basically you would only get this if you provide some object that does not conform to the typehint, i.e. not a QuantumCircuit. This is most likely what happens when it assumes its a QuantumCircuit and tries to access fields like num_qubits, that should be in the object, but are not. Python then raises this error. Arguably all bets are off if you pass the wrong types. I personally would remove this and the tests that pass some non QuantumCircuit object wherever this has been added.